src/socket/socket.cr

```diff
@@ -32,6 +32,48 @@ class Socket < IO::FileDescriptor
 
     def initialize(@family, @path)
     end
+
+    def initialize(sockaddr : LibC::SockAddrIn | LibC::SockAddrIn6)
+      @family = case sockaddr.family
+      when LibC::AF_INET then "AF_INET"
+      when LibC::AF_INET6 then "AF_INET6"
+      else raise "Unsupported address family"
```

Comment:
I'd love to make it use an enum. We already have `Socket::Family` which maps to the underlying C types. The only reason I kept a string was to preserve backwards compatibility with the existing usage of `Socket::Addr`. As long as it was documented as breaking change, would it be OK?